### PR TITLE
Allow forward seeking in caching stream

### PR DESCRIPTION
**Pull request by mtdowling**
@jeremeamia @jeskew

Addresses https://github.com/aws/aws-sdk-php/issues/587

### CODE REVIEW

File: src/CachingStream.php [hunk 1]
```diff
@@ -122,4 +130,12 @@ public function close()
     {
         $this->remoteStream->close() && $this->stream->close();
     }
+
+    private function cacheEntireStream()
```
Comment (jeremeamia) [inline src/CachingStream.php]: What exactly is this doing?

**Comment by jeskew** [comment]
The docblock still says a RuntimeException will be thrown if you seek from `SEEK_END`, but otherwise :shipit:

File: src/CachingStream.php (hunk 1 above)
Comment (mtdowling) [inline src/CachingStream.php]: This reads the entire stream, which ends up caching the entire stream.

**Comment by mtdowling** [comment]
Docblock fixed. Thanks.

**Pull request merged**